GliaWidgets/Sources/View/Call/Video/FlipCameraButton.swift

```diff
@@ -0,0 +1,126 @@
+import UIKit
+
+struct FlipCameraButtonStyle: Equatable {
+    struct State: Equatable {
+        let background: ColorType
+        let imageColor: ColorType
+    }
+
+    let activeState: State
+    let inactiveState: State
+    let selectedState: State
+}
+
+class FlipCameraButton: UIButton {
+    static let size = CGSize(width: 40, height: 40)
+    static let bottomTrailingPadding: CGFloat = -8
+
+    struct Props: Equatable {
+        let style: FlipCameraButtonStyle
+        let tap: Cmd?
+    }
+
+    var props = Props(
+        style: .init(
+            activeState: .nop,
+            inactiveState: .nop,
+            selectedState: .nop
+        ),
+        tap: nil
+    ) {
+        didSet {
+            renderProps()
+        }
+    }
+
+    private var renderedActiveState: FlipCameraButtonStyle.State = .nop {
+        didSet {
+            guard renderedActiveState != oldValue else {
```

Comment:
I don't understand, if the renderedActiveState starts with 0 let's say, and you change the value from 0 to 1, it means that renderedActiveState (0) is the same as oldValue (0), so it will return and not render the state 1.

Reply:
Yes, the idea is to avoid excessive re-rendering if value is the same, so we render if new value is different from old one. That is especially important, because we rely on background image generation with specific color for button background states and the same true for icon image used for button.

Reply:
Okay I understand now, `renderedActiveState` at this point already has the new value. For some reason when I read it, I thought that at this point it would have the old value, and thus checking the old value against `oldValue` didn't make much sense. Apologies.